ui/main/src/app/store/effects/card-operation.effects.ts

```diff
@@ -49,26 +51,47 @@ export class CardOperationEffects {
             map((loadedCardAction: LoadLightCard) => loadedCardAction.payload.lightCard),
             withLatestFrom(this.store.select(selectLightCardsState)),
             map(([lightCard, state]) => {
-                    if (!!lightCard.parentCardId) {
-                        return new LoadLightChildCard({lightCard: lightCard,isFromCurrentUserEntity : this.doesChildCardIsFromCurrentUserEntity(lightCard)});
-                    }
-                    else {
-                        let card: LightCard  = {...lightCard} ;
-                        card.hasChildCardFromCurrentUserEntity = this.doesNewCardVersionHasChildFromCurrentUserEntity(state.entities[card.id],card);
-                        return new LoadLightParentCard({lightCard: card});
+                if (!!lightCard.parentCardId) {
+                    const isFromCurrentUser = this.doesChildCardIsFromCurrentUserEntity(lightCard);
+                    if (isFromCurrentUser) {
+                        if (!state.entities[lightCard.parentCardId]) {
+                             // if parent does not exist yet keep in memory the information that the card with id=parentCardId has a child
+                             // and that this child if from current user entity
+                             // this situation can arise as we do not have the card ordered when received from
+                             // the back via subscription (i.e a child can be received  before his parent)
+                            this.orphanedChildCardsFromCurrentEntity.add(lightCard.parentCardId);
+                            // do not need to load this child card
+                            return new NoopAction();
+                        }
+    
                     }
+                    return new LoadLightChildCard({lightCard: lightCard, isFromCurrentUserEntity: isFromCurrentUser});
                 }
-            )
+
+                else {
+                    let card: LightCard = {...lightCard};
+                    card.hasChildCardFromCurrentUserEntity = this.doesCardHasChildFromCurrentUserEntity(state.entities[card.id], card);
```

Comment:
I think creating a variable for state.entities[card.id] (with the appropriate name, maybe currentLightCardState or something to that effect) rather than using it directly as a paramter would help code readability.
The "entities" naming (coming from ngrx entities) can be confusing in a method referring to "currentUserEntity".